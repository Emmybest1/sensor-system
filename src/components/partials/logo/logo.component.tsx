import React from 'react';
import { Link } from 'react-router-dom';
import { Paths } from 'routes';
import styled from 'styled-components';

const LogoWrapper = styled.div`
  font-size: 2rem;
  font-weight: 500;
`;

const BlueNode = styled.span`
  color: #3497fa;
`;

const GreenNode = styled.span`
  color: #76be76;
`;

const RedNode = styled.span`
  color: #f86c6c;
`;

const OrangeNode = styled.span`
  color: #ffa500;
`;

type LogoPropTypes = {
  noLink?: boolean;
};

export const Logo: React.FC<LogoPropTypes> = ({ noLink }) => {
  const logoElement = (
    <LogoWrapper>
      <GreenNode>S</GreenNode>
      <OrangeNode>e</OrangeNode>
      <RedNode>n</RedNode>
      <GreenNode>T</GreenNode>
      <BlueNode>e</BlueNode>
      <RedNode>m</RedNode>
    </LogoWrapper>
  );

  if (noLink) return logoElement;

  return <Link to={Paths.home}>{logoElement}</Link>;
};

Logo.defaultProps = {
  noLink: false,
};
