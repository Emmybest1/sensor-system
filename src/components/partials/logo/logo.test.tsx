import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Logo } from './logo.component';

describe('<Logo/>', () => {
  test('should render Component', () => {
    const { container } = render(
      <Router>
        <Logo />
        <Logo noLink />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
