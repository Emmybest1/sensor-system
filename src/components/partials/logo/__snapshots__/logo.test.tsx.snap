// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> should render Component 1`] = `
<a
  href="/"
>
  <div
    class="sc-fbkiRW gelFBc"
  >
    <span
      class="sc-fWSDcn csBAtW"
    >
      S
    </span>
    <span
      class="sc-TmdmN cyzjQY"
    >
      e
    </span>
    <span
      class="sc-dwfUuu fsFXTQ"
    >
      n
    </span>
    <span
      class="sc-fWSDcn csBAtW"
    >
      T
    </span>
    <span
      class="sc-GqePz hLSQzd"
    >
      e
    </span>
    <span
      class="sc-dwfUuu fsFXTQ"
    >
      m
    </span>
  </div>
</a>
`;
